### PR TITLE
fix: getter_return - checking the return value of the getter property in `Reflect.defineProperty()`

### PR DESCRIPTION
**Pull request by hirasawayuki**
# Fixes https://github.com/denoland/deno_lint/issues/1073

- Fixed to check the return value of get property in `Reflect.defineProperty`.
- Minimal test cases added because of common implementation with `Object.defineProperty`.

I would like to fix `Object.create()` separately when I have time to spare. (https://github.com/denoland/deno_lint/issues/1073#issuecomment-1227866555)

### CODE REVIEW

File: src/rules/getter_return.rs [hunk 1]
```diff
@@ -341,6 +341,8 @@ mod tests {
          { bar: { get: function() { return true; } } });"#,
       r#"Object.defineProperties(foo,
          { bar: { get: function () { ~function() { return true; }(); return true; } } });"#,
+      r#"Refrect.defineProperty(foo, "bar", { get: function () { return true; } });"#,
+      r#"Refrect.defineProperty(foo, "bar", { get: () => { return true; } });"#,
```
Comment (magurotuna) [inline src/rules/getter_return.rs:345]:
```suggestion
      r#"Reflect.defineProperty(foo, "bar", { get: function () { return true; } });"#,
      r#"Reflect.defineProperty(foo, "bar", { get: () => { return true; } });"#,
```

typo

**Review by magurotuna — CHANGES_REQUESTED** [review]
Good fix. I noticed just one typo :)

File: src/rules/getter_return.rs (hunk 1 above)
Comment (hirasawayuki) [reply src/rules/getter_return.rs]: Oh... Thank you for pointing it out 🙇‍♂️

**Review by magurotuna — APPROVED** [review]
LGTM 👍

**Pull request merged**